src/main/java/com/wasilewskyy/medical_clinic/repository/PatientRepository.java

```diff
@@ -32,12 +35,13 @@ public void deleteByEmail(String email) {
     }
 
     public void update(String email, Patient updatedPatient) {
-        for (int i = 0; i < patients.size(); i++) {
-            if (patients.get(i).getEmail().equals(email)) {
-                patients.set(i, updatedPatient);
-                return;
-            }
-        }
+        patients.stream()
+                .filter(patient -> patient.getEmail().equals(email))
+                .findFirst()
+                .ifPresent(patient -> {
+                    int index = patients.indexOf(patient);
```

Comment:
To jest podmiana jednej instancji na druga a nie modyfikacja istniejacego pacjenta
Powinienes znalezc Pacjenta ktoremu mamy zmienic dane i przy pomocy metod set ustawić mu te dane